File: leetcode-cn/2937.0_Make_Three_Strings_Equal.py
class Solution:
    def findMinimumOperations(self, s1: str, s2: str, s3: str) -> int:
        idx = -1
        for i, (a, b, c) in enumerate(zip(s1, s2, s3)):
            if a == b == c:
                idx = i
            else:
                break
        return -1 if idx == -1 else len(s1) + len(s2) + len(s3) - (idx + 1) * 3
    

class Solution:
    def findMinimumOperations(self, s1: str, s2: str, s3: str) -> int:
        lcp = 0
        for x, y, z in zip(s1, s2, s3):
            if x != y or x != z:
                break
            lcp += 1
        return -1 if lcp == 0 else len(s1) + len(s2) + len(s3) - lcp * 3

File: cpp/2937.0_Make_Three_Strings_Equal.cpp
class Solution {
public:
    int findMinimumOperations(string s1, string s2, string s3) {
        int n = min({s1.length(), s2.length(), s3.length()});
        int i = 0;
        while (i < n && s2[i] == s1[i] && s3[i] == s1[i]) {
            i++;
        }
        return i == 0 ? -1 : s1.length() + s2.length() + s3.length() - i * 3;
    }
};